File: flutter_obdii/lib/main_material.dart
import 'package:flutter/material.dart';
import 'package:flutter_localizations/flutter_localizations.dart';

import 'app_bootstrap.dart';
import 'theme/app_theme.dart';
import 'views/main_scaffold_material.dart';

Future<void> main() async {
  await bootstrapObdApp();
  runApp(const ObdMaterialApp());
}

class ObdMaterialApp extends StatelessWidget {
  const ObdMaterialApp({super.key});

  @override
  Widget build(BuildContext context) {
    return ObdAppProviders(
      child: MaterialApp(
        title: 'Rheosoft OBDII',
        debugShowCheckedModeBanner: false,
        theme: AppTheme.lightTheme,
        darkTheme: AppTheme.darkTheme,
        localizationsDelegates: const [
          GlobalMaterialLocalizations.delegate,
          GlobalWidgetsLocalizations.delegate,
          GlobalCupertinoLocalizations.delegate,
        ],
        supportedLocales: const [Locale('en', 'US')],
        themeMode: ThemeMode.system,
        home: const MainScaffold(),
      ),
    );
  }
}

File: kotlin_obdii/app/src/main/java/com/rheosoft/obdii/bootstrap/MainMaterial.kt
package com.rheosoft.obdii.bootstrap

import com.rheosoft.obdii.views.MainScaffold

data class MaterialAppSpec(
    val title: String,
    val debugShowCheckedModeBanner: Boolean,
    val useMaterial3: Boolean,
    val seedColorHex: String,
    val supportedLocales: List<String>,
    val themeMode: String,
    val home: String,
)

object MainMaterial {
    val appSpec = MaterialAppSpec(
        title = "Rheosoft OBDII",
        debugShowCheckedModeBanner = false,
        useMaterial3 = true,
        seedColorHex = "#00C2FF",
        supportedLocales = listOf("en_US"),
        themeMode = "system",
        home = MainScaffold::class.simpleName ?: "MainScaffold",
    )

    suspend fun run() {
        AppBootstrap.initialize()
    }
}
